### PR TITLE
[CLIP] fix logit_scale init

### LINKED ISSUE #13430: Difference between `logit_scale` initialisation in Transformers CLIP and the original OpenAI implementation.
I tried another training code based on the OpenAI'CLIP version: I found a difference at logit_scale between them. Does it mean temperature parameter? Is it the reason for loss rising?
huggingface transformers' CLIP:

```
self.logit_scale = nn.Parameter(torch.ones([]))
```
OpenAI CLIP:

```
self.logit_scale = nn.Parameter(torch.ones([]) * np.log(1 / 0.07))
```

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

This PR fixes the initialization of `logit_scale` parameter as per the [original CLIP implementation ](https://github.com/openai/CLIP/blob/3b473b0e682c091a9e53623eebc1ca1657385717/clip/model.py#L291).

Also, the initial value of `logit_scale` needs to be carefully tuned for CLIP training, so this PR, adds a `logit_scale_init_value` as a `config` parameter, so users could pass different init values as per their training.

Fixes #13430

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
Did we run a test training to verify that this gives better results?

**Review by sgugger — APPROVED** [review]
Thanks for fixing this (and +1 for the test)!
